mkosi/__init__.py

```diff
@@ -2255,3 +2270,27 @@ def run_verb(args: MkosiArgs, presets: Sequence[MkosiConfig]) -> None:
 
             if args.verb == Verb.serve:
                 run_serve(last)
+
+
+def normalize_mtime(root: Path, source_date_epoch: Optional[int]) -> None:
+    if source_date_epoch is None:
+        return
+    with complete_step("Setting mtime to SOURCE_DATE_EPOCH"):
+        os.utime(root, (source_date_epoch, source_date_epoch), follow_symlinks=False)
+        for p in root.rglob("*"):
+            os.utime(p, (source_date_epoch, source_date_epoch), follow_symlinks=False)
+
+
+def normalize_mtime_path(root: Path, target: Path, source_date_epoch: Optional[int]) -> None:
```

Comment:
Please merge those two functions like this:
```python
def normalize_mtime(source_date_epoch: Optional[int], root: Path, target: Optional[Path] = None):
    ...
```

You include `root` in the message when root is reset, but don't include `target` in the message. I think it should be the other way around. You should only print the path relative to `root`. When resetting the whole root, don't print the path, it's not interesting.